DeepSpeech.ipynb

```diff
@@ -1239,16 +1271,21 @@
     "        \n",
     "        # Validation step\n",
     "        if epoch % validation_step == 0:\n",
-    "            _, last_validation_wer = print_wer_report(session, \"Validation\", validation_tower_decodings, validation_tower_labels)\n",
+    "            _, last_validation_wer = print_wer_report( \\\n",
+    "                session, \\\n",
+    "                \"Validation\", \\\n",
+    "                validation_tower_decodings, \\\n",
+    "                validation_tower_labels, \\\n",
+    "                validation_tower_total_losses)\n",
     "            print\n",
     "\n",
     "        # Loop over the batches\n",
-    "        for batch in range(int(ceil(float(total_batches)/len(available_devices)))):\n",
+    "        for batch in range(int(ceil(float(total_batches) / len(available_devices)))):\n",
     "            # Compute the average loss for the last batch\n",
-    "            _, batch_avg_loss = session.run([apply_gradient_op, tower_loss], feed_dict_train)\n",
+    "            session.run(apply_gradient_op, feed_dict_train)\n",
     "\n",
     "            # Add batch to total_accuracy\n",
-    "            total_accuracy += session.run(accuracy, feed_dict_train)\n",
+    "            total_accuracy += session.run(tower_accuracies[-1], feed_dict_train)\n",
```

Comment:
This mimics the old behavior where just the last tower's accuracy was used. I agree with you on using the average of all tower accuracies instead.